packages/marketplace/src/components/ui/developer-analytics/billing/cost-explorer/cost-explorer.tsx

```diff
@@ -18,17 +23,35 @@ export const prepareFilterFormInitialValues = (createdMonth: string) => {
   }
 }
 
-export const handleOnSave = (setCreatedMonth: (createdMonth: string) => void) => {
+interface HandleOnSave {
+  setCreatedMonth: (createdMonth: string) => void
+  dispatch: Dispatch
+  myAppIds: string[]
+}
+
+export const handleOnSave = ({ setCreatedMonth, dispatch, myAppIds }: HandleOnSave) => {
   return (values: CostFilterFormValues) => {
     const { createdMonth } = values
     setCreatedMonth(createdMonth)
+    const month = dayjs(createdMonth).format(DATE_TIME_FORMAT.YYYY_MM)
+    dispatch(fetchMonthlyBilling({ month, applicationId: myAppIds }))
   }
 }
 
 const CostExplorer: React.FC<CostExplorerProps> = () => {
+  const dispatch = useDispatch()
+  const myApps = useSelector(selectDeveloperApps)
+  const myAppIds = myApps.map((item: AppSummaryModel) => item.id) as string[]
+
   const [createdMonth, setCreatedMonth] = React.useState(dayjs().format(DATE_TIME_FORMAT.YYYY_MM))
   const initialValues = React.useMemo(prepareFilterFormInitialValues(createdMonth), [createdMonth])
-  const onSave = React.useCallback(handleOnSave(setCreatedMonth), [])
+
+  React.useEffect(() => {
+    myApps.length && dispatch(fetchMonthlyBilling({ month: createdMonth, applicationId: myAppIds }))
+  }, [myApps.length])
+
+  const onSave = React.useCallback(handleOnSave({ setCreatedMonth, dispatch, myAppIds }), [myAppIds])
```

Comment:
As `onSave` function is passed as a callback to `CostFilterForm`, we should put `dispatch` into `useCallback` dependency to avoid rendering unnecessarily due to the changed reference.